### PR TITLE
DSCS-2477 adds maximum depth to getHierarchySummaryForWork and remove…

### PR DESCRIPTION
**Pull request by m-r-c**
…s child count to improve performance

@scoen @shuangzhou @ninhnguyen @jrixon @sjacob @weijunnla @adityaburra

### CODE REVIEW

**Comment by nlabuilder** [comment]
Build finished.

File: src/test/java/amberdb/AmberSessionTest.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void getHierarchySummaryForWorkShouldRetrieveAllAncestorsOfTheRequestedWo
 
         sessionWithLookups.commit();
 
-        List<WorkSummary> hierarchy = sessionWithLookups.getHierarchySummaryForWork(work.getId());
+        List<WorkSummary> hierarchy = sessionWithLookups.getHierarchySummaryForWork(work.getId(), null);
```
Comment (scoen) [inline src/test/java/amberdb/AmberSessionTest.java]: Suggestion: Could have a convenience function with the old signature that passes the null ? Something like the following so devs don't wonder about the null ?
```
public List<WorkSummary> getHierarchySummaryForWork(long workId) {
    return getHierarchySummaryForWork(long workId, null);
}
```

**Comment by scoen** [comment]
👍

File: src/test/java/amberdb/AmberSessionTest.java (hunk 1 above)
Comment (m-r-c) [inline src/test/java/amberdb/AmberSessionTest.java]: Done

**Comment by nlabuilder** [comment]
SonarQube analysis reported 3 issues

1. MAJOR [AmberSession.java](https://github.com/nla/amberdb/blob/a82a7ca87815fadda0ad3453b53bbbdff926ce2d/src/main/java/amberdb/AmberSession.java): This file has 1,098 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS00104)
1. MINOR [AmberSession.java#L324](https://github.com/nla/amberdb/blob/a82a7ca87815fadda0ad3453b53bbbdff926ce2d/src/main/java/amberdb/AmberSession.java#L324): Define a constant instead of duplicating this literal "collection" 3 times. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS1192)
1. MINOR [AmberSession.java#L359](https://github.com/nla/amberdb/blob/a82a7ca87815fadda0ad3453b53bbbdff926ce2d/src/main/java/amberdb/AmberSession.java#L359): Define a constant instead of duplicating this literal "bibId" 3 times. [rule](https://sonarqube.nla.gov.au/coding_rules#rule_key=squid%3AS1192)

**Comment by nlabuilder** [comment]
Build finished.

**Pull request merged**